README.md

```diff
@@ -1,2 +1,134 @@
+
+#### API Key and Secret Configuration
+
+```
+vonage.creds.api-key=your-api-key
+vonage.creds.secret=your-api-secret
+```
+
+This will give you access to automatically wire in the `VonageClient`, `AccountClient`, `ApplicationClient`, `ConversionClient`, `InsightClient`, `NumbersClient`, `RedactClient`, `SmsClient`, `SnsClient`, and `VerifyClient`.
+
+You can also provide a signature secret:
+
+```
+vonage.creds.api-key=your-api-key
+vonage.creds.signature=signature
+```
+
+#### Application ID and Private Key Configuration
+To gain access to the `VoiceClient` you will need to provide a [Vonage Application ID]() and location or contents of a private key file associated with the Vonage Application.
+
+Here is an example providing the path to the key:
+```
+vonage.creds.application-id=application-id
+vonage.creds.private-key-path=/path/to/your/private.key
+```
+
+Here is an example providing the contents of the key:
+```
+vonage.creds.application-id=application-id
+vonage.creds.private-key-contents=contents-of-the-key
+```
+
+### Customizing the `VonageClient`
+
+By default, the auto configuration will create a `VonageClient.Builder` using the information provided in your `application.properties`, `application.yml`, or other external configuration source. However, when registering the `VonageClient` and all of the other sub-clients, the auto configuration can opt to use a pre-registered `VonageClient.Builder`.
+
+For example, if you would like to customize the base URI that the library uses, you can register a custom version of the `VonageClient.Builder` in your configuration class:
+
+```java
+@Autowired
+VonageCredentialsProperties vonageCredentialsProperties;
+
+@Bean
+public VonageClient.Builder customvonageBuilder() {
+    return VonageClient.builder()
+            .apiKey(vonageCredentialsProperties.getApiKey())
+            .apiSecret(vonageCredentialsProperties.getSecret())
+            .httpConfig(HttpConfig.builder().baseUri("https://example.com").build());
+}
+```
+> Note that you must include your credentials as shown in this example. This builder completely replaces the automatically configured one.
+
+## Customize vonage Client Version
+
+By default, the Vonage Spring Boot Starter will transitively define Vonage Client to the latest version at its release. You can override this by adding a dependency on the Vonage Client, bringing in `4.2.0` for example:
+
+For Gradle:
+
+```groovy
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    implementation 'com.vonage:client:4.2.0'
+}
+```
+
+For Maven:
+
+```xml
+<dependency>
+  <groupId>com.vonage</groupId>
+  <artifactId>client</artifactId>
+  <version>4.2.0</version>
+</dependency>
+```
+
+Bringing in older versions of the supported Vonage Client, may result in some unforseen consequences and build errors. As a result here is a list of each version targeted by each version of the starter:
+
+| Vonage Spring Boot Starter | Vonage Java Client |
+|---|---|
+| v1.0.0 | v4.3.0 |
```

Comment:
There should only be 1 row entry in the table here. and that should be `| v1.0.0 | v5.5.0 |` since